ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java

```diff
@@ -0,0 +1,198 @@
+
+@ToolAnnotation(IndexMappingTool.NAME)
+public class IndexMappingTool implements Tool {
+    public static final String NAME = "IndexMappingTool";
+
+    private static final String DEFAULT_DESCRIPTION = "Use this tool to get index mapping information.";
+    @Setter
+    @Getter
+    private String name = IndexMappingTool.NAME;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+    @Getter
+    private String type;
+    @Getter
+    private String version;
+    private Client client;
+    @SuppressWarnings("unused")
+    private String modelId;
+    @Setter
+    private Parser<?, ?> inputParser;
+    @Setter
+    private Parser<?, ?> outputParser;
+    @SuppressWarnings("unused")
+    private ClusterService clusterService;
+
+    public IndexMappingTool(Client client, ClusterService clusterService, String modelId) {
+        this.client = client;
+        this.clusterService = clusterService;
+        this.modelId = modelId;
+
+        outputParser = new Parser<>() {
+            @Override
+            public Object parse(Object o) {
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        @SuppressWarnings("unchecked")
+        List<String> indexList = parameters.containsKey("index")
+            ? gson.fromJson(parameters.get("index"), List.class)
+            : Collections.emptyList();
+        if (indexList.isEmpty()) {
+            @SuppressWarnings("unchecked")
+            T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
+            listener.onResponse(empty);
+            return;
+        }
+        
+        final String[] indices = indexList.toArray(Strings.EMPTY_ARRAY);
+
+        final IndicesOptions indicesOptions = IndicesOptions.strictExpand();
+        final boolean local = parameters.containsKey("local") ? Boolean.parseBoolean("local") : false;
+        final TimeValue clusterManagerNodeTimeout = DEFAULT_CLUSTER_MANAGER_NODE_TIMEOUT;
+
+        ActionListener<GetIndexResponse> internalListener = new ActionListener<GetIndexResponse>() {
+
+            @Override
+            public void onResponse(GetIndexResponse getIndexResponse) {
+                try {
+                    // Handle empty response
+                    if (getIndexResponse.indices().length == 0) {
+                        @SuppressWarnings("unchecked")
+                        T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
+                        listener.onResponse(empty);
+                        return;
+                    }
+                    StringBuilder sb = new StringBuilder();
```

Comment:
Should we keep the output of tool follow json format and let the model understand from json format? this will help simplify each tool by directly returning json response

Reply:
Good question, the requirements for the output have not been specified and the format I gave was in response to my question for how it should be.  If we want a different format it needs to be clearly specified for me to write code against.  @ylwu-amzn @jngz-es

Reply:
> Should we keep the output of tool follow json format and let the model understand from json format

Unless we do see the benefit of this, I would suggest not enforcing this. Keep it flexible and we could fine tune later.

> the requirements for the output have not been specified and the format I gave was in response to my question for how it should be

It's hard to tell what's the exact format which will works the best without testing and fine tuning. Let's not worry too much about format for now. We can iterate fast and fine tune later.

Reply:
Agree with moving forward on dev branch. However, I would prefer to define a unified format for all tool output so that it is easier for other tool development. If we are inclined on moving tuning the output later, I would prefer tools output in json format and let agent implement a common mechanism to covert from json to other formats.